handler/src/test/java/io/netty/handler/ssl/RenegotiateTest.java

```diff
@@ -0,0 +1,132 @@
+
+public abstract class RenegotiateTest {
+
+    @Test(timeout = 5000)
+    public void testRenegotiateServer() throws Exception {
+        final CountDownLatch latch = new CountDownLatch(2);
+        SelfSignedCertificate cert = new SelfSignedCertificate();
+        EventLoopGroup group = new LocalEventLoopGroup();
+        try {
+            final SslContext context = SslContextBuilder.forServer(cert.key(), cert.cert())
+                    .sslProvider(serverSslProvider()).build();
+            ServerBootstrap sb = new ServerBootstrap();
+            sb.group(group).channel(LocalServerChannel.class)
+                    .childHandler(new ChannelInitializer<Channel>() {
+                        @Override
+                        protected void initChannel(Channel ch) throws Exception {
+                            ch.pipeline().addLast(context.newHandler(ch.alloc()));
+                            ch.pipeline().addLast(new ChannelInboundHandlerAdapter() {
+                                private boolean renegotiate;
+
+                                @Override
+                                public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception {
+                                    ReferenceCountUtil.release(msg);
+                                }
+
+                                @Override
+                                public void userEventTriggered(
+                                        final ChannelHandlerContext ctx, Object evt) throws Exception {
+                                    if (!renegotiate && evt instanceof SslHandshakeCompletionEvent) {
+                                        SslHandshakeCompletionEvent event = (SslHandshakeCompletionEvent) evt;
+
+                                        if (event.isSuccess()) {
+                                            final SslHandler handler = ctx.pipeline().get(SslHandler.class);
+
+                                            renegotiate = true;
+                                            handler.engine().setEnabledCipherSuites(
+                                                    new String[]{"SSL_RSA_WITH_3DES_EDE_CBC_SHA"});
+                                            handler.renegotiate().addListener(new FutureListener<Channel>() {
+                                                @Override
+                                                public void operationComplete(Future<Channel> future) throws Exception {
+                                                    if (!future.isSuccess()) {
+                                                        future.cause().printStackTrace();
+                                                        ctx.close();
+                                                    }
+                                                }
+                                            });
+                                        } else {
+                                            event.cause().printStackTrace();
```

Comment:
can we propagate this to throw from the test so it reports in junit more cleanly?